LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionCN470RP2.cs

```diff
@@ -280,46 +302,32 @@ public override RX2ReceiveWindow GetDefaultRX2ReceiveWindow(DeviceJoinInfo devic
                 // 20 MHz plan A
                 if (deviceJoinInfo.DesiredCN470JoinChannel <= 7)
                 {
-                    return new RX2ReceiveWindow(486.9, dataRate);
+                    return new RX2ReceiveWindow(Mega(486.9), dataRate);
                 }
                 // 20 MHz plan B
                 else if (deviceJoinInfo.DesiredCN470JoinChannel <= 9)
                 {
-                    return new RX2ReceiveWindow(498.3, dataRate);
+                    return new RX2ReceiveWindow(Mega(498.3), dataRate);
                 }
                 // 26 MHz plan A
                 else if (deviceJoinInfo.DesiredCN470JoinChannel <= 14)
                 {
-                    return new RX2ReceiveWindow(492.5, dataRate);
+                    return new RX2ReceiveWindow(Mega(492.5), dataRate);
                 }
                 // 26 MHz plan B
                 else if (deviceJoinInfo.DesiredCN470JoinChannel <= 19)
                 {
-                    return new RX2ReceiveWindow(502.5, dataRate);
+                    return new RX2ReceiveWindow(Mega(502.5), dataRate);
                 }
             }
 
             return rx2Window;
         }
 
-        private static List<double> BuildFrequencyPlanList(double startFrequency, int startChannel, int endChannel)
-        {
-            var frequencies = new List<double>();
-            var currentFreq = startFrequency;
-
-            for (var channel = startChannel; channel <= endChannel; ++channel)
-            {
-                frequencies.Add(Math.Round(currentFreq, 1, MidpointRounding.AwayFromZero));
-                currentFreq += FrequencyIncrement;
-            }
-
-            return frequencies;
-        }
-
         [Obsolete("#655 - This Rxpk based implementation will go away as soon as the complete LNS implementation is done.")]
         private static int GetChannelNumber(Rxpk upstreamChannel, double startUpstreamFreq, int startChannelNumber = 0) =>
-            startChannelNumber + (int)Math.Round((upstreamChannel.Freq - startUpstreamFreq) / FrequencyIncrement, 0, MidpointRounding.AwayFromZero);
-        private static int GetChannelNumber(double upstreamChannelFrequency, double startUpstreamFreq, int startChannelNumber = 0) =>
-            startChannelNumber + (int)Math.Round((upstreamChannelFrequency - startUpstreamFreq) / FrequencyIncrement, 0, MidpointRounding.AwayFromZero);
+            startChannelNumber + (int)Math.Round((upstreamChannel.Freq - startUpstreamFreq) / FrequencyIncrement.Value, 0, MidpointRounding.AwayFromZero);
+        private static int GetChannelNumber(Hertz upstreamChannelFrequency, Hertz startUpstreamFreq, int startChannelNumber = 0) =>
+            startChannelNumber + (int)Math.Round((upstreamChannelFrequency.InMega - startUpstreamFreq.InMega) / FrequencyIncrement.Value, 0, MidpointRounding.AwayFromZero);
```

Comment:
This is correct but I think we could do the same as in RegionCN470RP1
```
private static int GetChannelNumber(Hertz upstreamChannelFrequency, Hertz startUpstreamFreq, int startChannelNumber = 0) =>
startChannelNumber + (int)Math.Round((upstreamChannelFrequency - startUpstreamFreq) / FrequencyIncrement.Units, 0, MidpointRounding.AwayFromZero);
```

Reply:
Sorry, but I am confused. There is no getting of channel number in `RegionCN470RP1`.

Reply:
Okay, I think I got it now. You mean the channel number math in `TryGetDownstreamChannelFrequency` of `RegionCN470RP1`.  Done in ced2959e2ac67572cc3dbceae23d1252c870a59b.